dataset/instrution_commit_clean.py

```diff
@@ -0,0 +1,266 @@
+        new_file_end = max(file2_range[1], new_file_end)
+
+    # -2 for compatibility with gh_diff
+    example["old_change_start"] = old_file_start
+    example["old_change_end"] = old_file_end
+    example["old_change_range"] = old_file_end - old_file_start
+
+    example["new_change_start"] = new_file_start
+    example["new_change_end"] = new_file_end
+    example["new_change_range"] = new_file_end - new_file_start
+
+    example["n_inserts"] = n_inserts
+    example["n_deletes"] = n_deletes
+    example["n_changes"] = n_inserts + n_deletes
+
+    return example
+
+
+counter = Value('i', 0)
+
+
+def init(args):
+    ''' store the counter for later use '''
+    global counter
+    counter = args
+
+
+def prepare_download_files():
+    downloaded_files = []
+    for i in range(1, 459):
+        downloaded_files.append("data/python/python-{:04d}.jsonl".format(i))
+
+    for i in range(1, 517):
+        downloaded_files.append("data/javascript/javascript-{:04d}.jsonl".format(i))
+
+    for i in range(1, 250):
+        downloaded_files.append("data/java/java-{:04d}.jsonl".format(i))
+    return downloaded_files
+
+
+data_files = prepare_download_files()
+
+
+def download_file(file):
+    global counter
+    print("start")
+    file = hf_hub_download(DATASET_NAME, file, repo_type="dataset",
+                           cache_dir=CACHE_DIR)
+    with counter.get_lock():
+        counter.value += 1
+    print(counter.value)
+    return file
+
+
+# download files using multi-thread
+with Pool(16, initializer=init, initargs=(counter,)) as p:
+    _ = p.map(download_file, data_files)
+
+# obtain the file path
+files = [hf_hub_download(DATASET_NAME, file, repo_type="dataset",
+                         cache_dir=CACHE_DIR) for file in data_files]
+
+counter = Value('i', 0)
+
+
+def load_file(file):
+    global counter
+    print("start")
+    file = load_dataset("/".join(file.split("/")[:-1]), data_files=file,
+                        split="train", cache_dir=CACHE_DIR)
+    with counter.get_lock():
+        counter.value += 1
+    print(counter.value)
+    return file
+
+
+with Pool(8, initializer=init, initargs=(counter,)) as p:
+    ds_list = p.map(load_file, files)
+
+ds = concatenate_datasets(ds_list)
+
+print("The dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: x["proba"] >= 0.9, num_proc=30)
+
+print("After proba filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.filter(lambda x: len(x["old_contents"]) < 100_000, num_proc=30)
+
+print("After content length filtering, the dataset size is: {}".format(len(ds)))
+
+ds = ds.map(get_line_diff_range, num_proc=30)
+
+
+def commit_filter(example):
+    if len(example["new_contents"]) == 0:
+        return False
+
+    if example["old_change_range"] <= 2:
+        if random.random() > SHORT_SAMPLING:
+            return False
+
+    if example["old_change_range"] >= 200:
+        if random.random() > LONG_SAMPLING:
+            return False
```

Comment:
This step removes around 25% in my experiments on a sample of 10K, things such as below:

```
['Implement code-tests for the following checks:\n\n* com.google.fonts/check/081\n* com.google.fonts/check/083\n* com.google.fonts/check/084\n', 'refactor\n', 'weather: need to return None if date_period is out of YahooWeather data range\n', 'Eliminate redundant options for cylindrical/disc flames\n', 'var name typo: n_gramming\n', 'Replace `optparse` with `argparse`\n', 'tgrep: string literals to unicode\n', 'indent with four spaces\n', "use desired type instead of 'Any'\n", 'Clean up whitespace.\n']
```

Upping the range to 500, would reduce it to only 50%. Maybe worth running an ablation?

Reply:
Yes it should be an option. I will explore it later!